src/pheval/post_processing/post_processing.py

```diff
@@ -156,20 +162,18 @@ def _return_sort_order(sort_order_str: str) -> SortOrder:
         raise ValueError("Incompatible ordering method specified.")
 
 
-def _create_pheval_result(
-    pheval_result: [PhEvalGeneResult] or [PhEvalVariantResult], sort_order_str: str
-) -> [RankedPhEvalGeneResult] or [RankedPhEvalVariantResult]:
-    """Create PhEval gene/variant result with corresponding ranks."""
+def _create_pheval_result(pheval_result: [PhEvalResult], sort_order_str: str) -> [PhEvalResult]:
+    """Create PhEval result with corresponding ranks."""
     sort_order = _return_sort_order(sort_order_str)
     sorted_pheval_result = ResultSorter(pheval_result, sort_order).sort_pheval_results()
     return _rank_pheval_result(sorted_pheval_result, sort_order)
 
 
 def _write_pheval_gene_result(
-    ranked_pheval_result: [RankedPhEvalGeneResult], output_dir: Path, tool_result_path: Path
+    ranked_pheval_result: [PhEvalResult], output_dir: Path, tool_result_path: Path
 ) -> None:
     """Write ranked PhEval gene result to tsv."""
-    ranked_result = pd.DataFrame([x.as_dict() for x in ranked_pheval_result])
+    ranked_result = pd.DataFrame([data.__dict__ for data in ranked_pheval_result])
```

Comment:
Why this change?

Reply:
I ended up removing the method `as_dict()` method in the class as Jules wanted me to avoid creating those dictionary/dictionary keys manually